src/XMakeCommandLine/XMake.cs

```diff
@@ -1363,6 +1357,30 @@ internal static void GatherCommandLineSwitches(ArrayList commandLineArgs, Comman
         }
 
         /// <summary>
+        /// If on Unix, check if the string looks like a file path.
+        /// The heuristic is if something resembles paths (contains slashes) check if the
+        /// first segment exists and is a directory.
+        /// </summary>
+        private static bool LooksLikeUnixFilePath(string value)
+        {
+#if MONO
+            var firstSlash = value.IndexOf('/');
```

Comment:
Can this be factored out over here as a helper?https://github.com/Microsoft/msbuild/blob/xplat/src/Shared/FileUtilities.cs#L475-L485